### DISCUSSION TITLE
Emitting WASM binary using C API header file

### DISCUSSION THREAD

**Discussion by ubaidsk** [WebAssembly/binaryen#4597] [Q&A]
In the C API, there is no function which could emit WebAssembly in text format (there is a function named `BinaryenModulePrint` which just prints the text format). I made the following function which emits the webassembly code (it is just a temporary hack which uses the overloaded `cout`)

```
const char* BinaryenModuleEmitText(BinaryenModuleRef module) {
  std::ostream out(std::cout.rdbuf());
  out << (*(Module*)module);
  std::stringstream ss;
  ss << out.rdbuf();
  return ss.str().c_str();
}
```

I needed the `wasm` binary form of the WebAssembly Text. Is there any function which prints the binary format or emits it? Or is there any function which could be modified to emit binary format? Or is there a way in the C API library to convert the WebAssembly text format to `wasm` binary format?

**Comment by tlively** [comment]
It sounds like `BinaryenModuleWrite` and `BinaryenModuleWriteText` might do what you want.

**Reply by ubaidsk** [reply]
@tlively got it. Thank you so much!
